classes/controller/menu/class-blocks-usage-controller.php

```diff
@@ -199,18 +276,31 @@ public function plugin_blocks_report() {
 				]
 			);
 			$results = $wpdb->get_results( $prepared_sql );
-			echo '<hr>';
-			echo '<h2>Tags that use a redirection page</h2>';
-			echo '<table><tr style="text-align: left"><th>ID</th><th>Title</th></tr>';
-			foreach ( $results as $result ) {
-				echo '<tr><td>'.$result->term_id .'</td>';
-				echo '<td><a href="term.php?taxonomy=post_tag&tag_ID=' . $result->term_id . '" >' . $result->name . '</a></td></tr>';
+			if ( 'text' === $type ) {
+				echo '<hr>';
+				echo '<h2>Tags that use a redirection page</h2>';
+				echo '<table><tr style="text-align: left">
+					<th>' . __( 'ID', 'planet4-blocks-backend' ) . '</th>
+					<th>' . __( 'Title', 'planet4-blocks-backend' ) . '</th>
+				</tr>';
+				foreach ( $results as $result ) {
+					echo '<tr><td>'.$result->term_id .'</td>';
+					echo '<td><a href="term.php?taxonomy=post_tag&tag_ID=' . $result->term_id . '" >' . $result->name . '</a></td></tr>';
+				}
+				echo '</table>';
+			} else {
+				$report[ 'TagsUsingRedirectionPage' ] = count($results);
 			}
-			echo '</table>';
 			// phpcs:enable
+
+			if ( 'json' === $type ) {
+				return $report;
+			}
 		}
 
 		/**
+		 * Return All WP block types(P4+Core).
```

Comment:
Perhaps I should have added a comment here when I wrote this function. But maybe you can adjust it to say that it only returns some core block types, not all of them.